File: backend/PortfolioApi/Models/ContactMessage.cs
using System.ComponentModel.DataAnnotations;

namespace PortfolioAPI.Models;

public sealed class ContactMessage
{
    public int Id { get; set; }

    [Required, MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [Required, MaxLength(255), EmailAddress]
    public string Email { get; set; } = string.Empty;

    [Required, MaxLength(180)]
    public string Subject { get; set; } = string.Empty;

    [Required, MaxLength(80)]
    public string InquiryType { get; set; } = string.Empty;

    [Required, MaxLength(2000)]
    public string Message { get; set; } = string.Empty;

    [MaxLength(64)]
    public string? IpAddress { get; set; }

    [MaxLength(512)]
    public string? UserAgent { get; set; }

    public DateTime SubmittedAtUtc { get; set; } = DateTime.UtcNow;
    public bool IsRead { get; set; }
    public bool EmailNotificationSent { get; set; }
    [MaxLength(1000)]
    public string? EmailFailureReason { get; set; }
}

File: backend/portfolio-api/src/main/java/com/nikhilwabale/portfolioapi/entity/ContactMessage.java
package com.nikhilwabale.portfolioapi.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.OffsetDateTime;

/**
 * Maps to the existing "ContactMessages" table created by the previous EF Core backend.
 * Column names are quoted/case-preserved via hibernate.globally_quoted_identifiers
 * (see application.yml) so the schema is reused as-is, unmodified.
 */
@Entity
@Table(name = "ContactMessages")
@Getter
@Setter
@NoArgsConstructor
public class ContactMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "Id")
    private Integer id;

    @Column(name = "Name", length = 100, nullable = false)
    private String name;

    @Column(name = "Email", length = 255, nullable = false)
    private String email;

    @Column(name = "Subject", length = 180, nullable = false)
    private String subject;

    @Column(name = "InquiryType", length = 80, nullable = false)
    private String inquiryType;

    @Column(name = "Message", length = 2000, nullable = false)
    private String message;

    @Column(name = "IpAddress", length = 64)
    private String ipAddress;

    @Column(name = "UserAgent", length = 512)
    private String userAgent;

    @Column(name = "SubmittedAtUtc", nullable = false)
    private OffsetDateTime submittedAtUtc = OffsetDateTime.now(java.time.ZoneOffset.UTC);

    // Reserved for a future admin "mark as read" view; not set by the contact flow itself.
    // Mirrors the old C# entity, where IsRead was equally unused by ContactController.
    @Column(name = "IsRead", nullable = false)
    private boolean read = false;

    @Column(name = "EmailNotificationSent", nullable = false)
    private boolean emailNotificationSent = false;

    @Column(name = "EmailFailureReason", length = 1000)
    private String emailFailureReason;
}
